diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..181b450
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.1)
+project(cuda-lane-detection)
+
+find_package(OpenCV REQUIRED)
+find_package(CUDA 10 REQUIRED)
+message("-- CUDA version: ${CUDA_VERSION}")
+
+set(
+	CUDA_NVCC_FLAGS
+	${CUDA_NVCC_FLAGS}; 
+    -O3 
+	-gencode arch=compute_62,code=sm_62
+)
+
+file(GLOB projectSources *.cpp *.cu)
+
+cuda_add_executable(${CMAKE_PROJECT_NAME} ${projectSources})
+target_link_libraries(${CMAKE_PROJECT_NAME} ${OpenCV_LIBS})
+
+# nvcc -gencode arch=compute_62,code=sm_62 
+# -I/usr/local/include/opencv4 
+# -L/usr/local/lib/ *.cpp *cu -lopencv_core -lopencv_highgui -lopencv_imgcodecs -lopencv_imgproc -lopencv_videoio -std=c++11 -o $root/bin/cudaLaneDetection
diff --git a/commons.h b/commons.h
index dcbf215..17e874e 100644
--- a/commons.h
+++ b/commons.h
@@ -1,11 +1,11 @@
 #ifndef __COMMONS_H__					// avoid repeated expansion
 #define __COMMONS_H__
 
-#include <opencv2/core.hpp>
-#include <opencv2/imgcodecs.hpp>
-#include <opencv2/imgproc.hpp>
-#include <opencv2/highgui.hpp>
-#include "opencv2/videoio.hpp"
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgcodecs/imgcodecs.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include "opencv2/highgui/highgui.hpp"
 #include <iostream>
 #include <vector>
 #include <stdio.h>
